src/libLookup/Lookup.cpp

```diff
@@ -1642,17 +1708,417 @@ bool Lookup::InitMining()
     {
         LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
                   "I have successfully join the network");
-#ifndef IS_LOOKUP_NODE
         LOG_GENERAL(INFO, "Clean TxBodyDB except the last one");
         int size_txBodyDBs
             = (int)BlockStorage::GetBlockStorage().GetTxBodyDBSize();
         for (int i = 0; i < size_txBodyDBs - 1; i++)
         {
             BlockStorage::GetBlockStorage().PopFrontTxBodyDB(true);
         }
+    }
+
+    return true;
+}
+#endif // IS_LOOKUP_NODE
+
+bool Lookup::ProcessSetLookupOffline(const vector<unsigned char>& message,
+                                     unsigned int offset, const Peer& from)
+{
+    LOG_MARKER();
+#ifdef IS_LOOKUP_NODE
+    if (IsMessageSizeInappropriate(message.size(), offset, sizeof(uint32_t)))
+    {
+        return false;
+    }
+
+    // 4-byte listening port
+    uint32_t portNo
+        = Serializable::GetNumber<uint32_t>(message, offset, sizeof(uint32_t));
+    offset += sizeof(uint32_t);
+
+    uint128_t ipAddr = from.m_ipAddress;
+    Peer requestingNode(ipAddr, portNo);
+
+    {
+        lock_guard<mutex> lock(m_mutexOfflineLookups);
+        auto iter = std::find(m_lookupNodes.begin(), m_lookupNodes.end(),
+                              requestingNode);
+        if (iter != m_lookupNodes.end())
+        {
+            m_lookupNodesOffline.push_back(*iter);
```

Comment:
Can we not use pointer here?

Reply:
it's the dereference of the iterator

Reply:
Will there be any memory issue down the line? Will it be better or the same if I push_back requestingNode instead of *iter